src/Tag/StorageInterface.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace FOS\HttpCache\Tag;
+
+/**
+ * Implementors of this interface associate cache entry identifiers
+ * with tags, retrieve cache entry identifiers for tags and remove tags.
+ */
+interface StorageInterface
+{
+    /**
+     * Associate a list of tags with the given cache identifier.
+     * The expriry represents the TIL for the cache entry, it might
+     * be the value of the S-MAXAGE header for example.
+     *
+     * The identifier can be any scalar value which can be associated with a
+     * unique HTTP cache entry.
+     *
+     * @param string[] $tags
+     * @param mixed $identifier
+     * @param integer $expiry
+     * @return void
+     */
+    public function tagCacheId(array $tags, $identifier, $expiry = null);
```

Comment:
This method would typically be a proxy method.

The reason for the separation is in the other method: `invalidateTags`. The Symfony Tag Manager *knows* about the Symfony HTTP cache, it is responsible for removing the cache entry. The `StorageInterface` is agnostic to the cache implementations.

But I wonder now if we should remove the `ManagerInterface` and use the `StorageInterface` directly in the class which actually tags the cache entries (in this case the `TagSubscriber`) -- BUT the problem then is that you could have 2 separate implementations of the StorageInterface being used instead of 1.

Having this proxy method ensures that there is a single API (although the `TagSubscriber` currently only uses the `tagCacheId` method, it could/will also need access to `invalidateTags` in the case that invalidation is done via. a HTTP request).